src/main/java/at/codersbay/datastructures/Fractional.java

```diff
@@ -5,30 +5,69 @@
  * e.g.: 1/2, 1/3, 24/37, ..
  */
 public class Fractional {
+	
+	private final Integer numerator;
+	private final Integer denominator;
 
-    public Fractional(Integer numerator, Integer denominator) {
-        // TODO implement to fix tests in FractionalTest1
+    public Fractional(Integer numerator, Integer denominator) throws IllegalArgumentException {
+    	if (numerator == null || denominator == null) {
+    		throw new IllegalArgumentException("You shall not pass null!");
+    	} else if (denominator == 0) {
+			throw new IllegalArgumentException("You shall not divide through zero!");
+		} else {
+			this.numerator = numerator;
+			this.denominator = denominator;
+		}
     }
 
     public Integer getNumerator() {
-        // TODO implement to fix tests in FractionalTest1
-        return -1;
+        return numerator;
     }
 
     public Integer getDenominator() {
-        // TODO implement to fix tests in FractionalTest1
-        return -1;
+        return denominator;
     }
 
     public Float asFloat() {
-        // TODO implement to fix tests in FractionalTest2
-        return -1f;
+    	Float result = ((float) this.numerator / (float) this.denominator);
+        return result;
     }
 
     @Override
-    public boolean equals(Object obj) {
-        // TODO implement to fix tests in FractionalTest3
-        return false;
+	public boolean equals(Object obj) {
+		// TODO handle cases where a non-fractional is passed as an argument
```

Comment:
you can use the `instanceOf` operator to check whether obj is a Fractional or not and if not you can just return false